pkg/reconciler/broker/broker.go

```diff
@@ -272,42 +233,14 @@ func (r *Reconciler) reconcile(ctx context.Context, b *v1alpha1.Broker) error {
 
 	// So, at this point the Broker is ready and everything should be solid
 	// for the triggers to act upon.
-	err = r.reconcileTriggers(ctx, b, filterSvc)
-	if err != nil {
-		logging.FromContext(ctx).Error("Problem reconciling triggers", zap.Error(err))
-		return fmt.Errorf("failed to reconcile triggers: %v", err)
-	}
-	return nil
+	return filterSvc, nil
 }
 
-func (r *Reconciler) updateStatus(ctx context.Context, desired *v1alpha1.Broker) (*v1alpha1.Broker, error) {
-	broker, err := r.brokerLister.Brokers(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
-
-	// If there's nothing to update, just return.
-	if reflect.DeepEqual(broker.Status, desired.Status) {
-		return broker, nil
+func (r *Reconciler) FinalizeKind(ctx context.Context, b *v1alpha1.Broker) pkgreconciler.Event {
+	if err := r.propagateBrokerStatusToTriggers(ctx, b.Namespace, b.Name, nil); err != nil {
```

Comment:
Does this need the class check too?

Reply:
class is not in the repo yet

Reply:
I was basing my comment on the block just inside ReconcileKind :)

https://github.com/knative/eventing/pull/2549/files#diff-99d1987d2919a2cce497502d2ecee2b0R111

Reply:
I'll add the class checks in another follow on PR? Cool?

Reply:
Works for me :)